protocols/mdns/src/service.rs

```diff
@@ -213,6 +206,36 @@ impl $service_name {
     // resolves, not forcing self-referential structures on the caller.
     pub async fn next(mut self) -> (Self, MdnsPacket) {
         loop {
+            while let Some(event) = self.if_watch.next().now_or_never() {
```

Comment:
Could we perhaps better integrate the `if_watch.next()` future into the [select block that waits for the next inbound datagram or the next query interval](https://github.com/libp2p/rust-libp2p/blob/36c997c7e305baabd714c5d5a571ed71bc58ec13/protocols/mdns/src/service.rs#L273-L281), e.g. by combining it in another `select()` with `query_interval.next()`? It would avoid potential unnecessary delays in receiving `IfEvent::Up` while waiting for the query interval to elapse and there is no inbound traffic.

Reply:
Done